qiskit_experiments/base_analysis.py

```diff
@@ -87,33 +89,54 @@ def run(
         analysis_options = analysis_options.__dict__
 
         # Run analysis
+        analysis_result_parameters = {
+            "result_type": experiment_data.experiment_type,
```

Comment:
This will probably have to change, because some experiments (e.g. T2Ramsey) will have multiple analysis results. Probably the result_data will contain the type, set in _run_analysis, as already done today in the curve analysis class in the main branch.

Reply:
To add to this, result type + experiment id + device component must be unique in the results DB. So if an experiment has multiple analysis results, then they'd already have the same type + id. If they also have the same device components the save would fail.

Reply:
Sounds like we need to add sub-digit to the experiment id? Something like `xxxxx.y` with something like sub-net mask (`11111.0`) to easily find common experiment id?

Reply:
I think `result_type` should be set to `res["result_type"]` if this key exists, and otherwise to `experiment_type`. And it will be the experiment's responsibility to create multiple analysis results with different result types.